src/Ionide.ProjInfo/Library.fs

```diff
@@ -645,38 +658,67 @@ module ProjectLoader =
 
             Error(exc.Message)
 
-    let getFscArgs (LoadedProject project) =
-        project.Items
+    let getFscArgs (p: ProjectInstance) =
+        p.Items
         |> Seq.filter (fun p -> p.ItemType = "FscCommandLineArgs")
         |> Seq.map (fun p -> p.EvaluatedInclude)
 
-    let getCscArgs (LoadedProject project) =
-        project.Items
+    let getCscArgs (p: ProjectInstance) =
+        p.Items
         |> Seq.filter (fun p -> p.ItemType = "CscCommandLineArgs")
         |> Seq.map (fun p -> p.EvaluatedInclude)
 
-    let getP2PRefs (LoadedProject project) =
-        project.Items
-        |> Seq.filter (fun p -> p.ItemType = "_MSBuildProjectReferenceExistent")
-        |> Seq.map (fun p ->
-            let relativePath = p.EvaluatedInclude
-            let path = p.GetMetadataValue "FullPath"
+    let getP2PRefs (project) =
+        match project with
+        | TraversalProject p ->
+            let references =
+                p.Items
+                |> Seq.filter (fun p -> p.ItemType = "ProjectReference")
+                |> Seq.toList
+
+            let mappedItems = ResizeArray()
+
+            for item in references do
+                let relativePath = item.EvaluatedInclude
+                let fullPath = Path.GetFullPath(relativePath, item.Project.Directory)
+
+                {
+                    RelativePath = relativePath
+                    ProjectFileName = fullPath
+                    TargetFramework = ""
+                }
+                |> mappedItems.Add
 
-            let tfms =
-                if p.HasMetadata "TargetFramework" then
-                    p.GetMetadataValue "TargetFramework"
+            Seq.toList mappedItems
+
+        | StandardProject p ->
+            p.Items
+            |> Seq.choose (fun p ->
+                if
+                    p.ItemType
+                    <> "_MSBuildProjectReferenceExistent"
+                then
+                    None
                 else
-                    p.GetMetadataValue "TargetFrameworks"
+                    let relativePath = p.EvaluatedInclude
+                    let path = p.GetMetadataValue "FullPath"
 
-            {
-                RelativePath = relativePath
-                ProjectFileName = path
-                TargetFramework = tfms
-            }
-        )
+                    let tfms =
+                        if p.HasMetadata "TargetFramework" then
+                            p.GetMetadataValue "TargetFramework"
+                        else
+                            p.GetMetadataValue "TargetFrameworks"
 
-    let getCompileItems (LoadedProject project) =
-        project.Items
+                    Some {
+                        RelativePath = relativePath
+                        ProjectFileName = path
+                        TargetFramework = tfms
+                    }
+            )
+            |> Seq.toList
```

Comment:
The only logical operation that we need for Traversal Projects is finding ProjectReferences, which is pretty much just a lookup for `ProjectReference` items. Unlike in a 'normal' project, these are able to be used because we don't really have to handle the intricacies of the ProjectReference protocol